csharp/private/actions/assembly.bzl

```diff
@@ -155,14 +155,16 @@ def AssemblyAction(
     args.set_param_file_format("multiline")
     args.use_param_file("@%s")
 
+    direct_inputs = srcs + resources + analyzer_assemblies + additionalfiles + [toolchain.compiler]
+    direct_inputs += [keyfile] if keyfile else []
```

Comment:
@jimevans so the problem was that it works on Windows because no sandbox, but on other platforms it failed. So we need to tell the action that the key is an input to the compiler and it works everywhere.